### PR TITLE
Fix: `get_edit_term_link()` optional taxonomy parameter error

### PR DESCRIPTION
**Pull request by Debarghya-Banerjee**
Trac Ticket: [Core-61726](https://core.trac.wordpress.org/ticket/61726)

## Problem

- The `get_edit_term_link()` function currently generates an incorrect URL when the `$taxonomy` parameter is omitted. This issue arises because `$taxonomy` was made optional. However, the implementation overlooked consistently using `$tax->name` to populate the taxonomy query parameter in all scenarios.

## Solution

- To resolve this issue, we have updated `get_edit_term_link()` to consistently use `$tax->name` instead of `$taxonomy` when generating the URL. This ensures that the correct taxonomy name is always included in the URL structure, preventing errors and inconsistencies.

## Steps to Reproduce

- Call `get_edit_term_link( 42 )` where `42` is the `ID` of a term.
    - Without the fix, observe that the generated URL lacks the taxonomy parameter, leading to incorrect behavior.

## Changes Made

- Modified `get_edit_term_link()` to use `$tax->name` for populating the taxonomy query parameter.
- Added a test case to verify that the issue is resolved and the correct URL structure is maintained.

## Testing

- Tested the fix by invoking `get_edit_term_link()` with and without the $taxonomy parameter.
- Verified that the URL structure is correct in both cases and no errors occur.
- This enhancement ensures consistent behavior in `get_edit_term_link()` when determining the correct taxonomy for generating term edit URLs.

### CODE REVIEW

File: tests/phpunit/tests/link/getEditTermLink.php [hunk 1]
```diff
@@ -196,6 +196,25 @@ function ( $location, $term ) {
...
+	 *
+	 * @return void
+	 */
+	public function test_get_edit_term_link_without_taxonomy() {
+		$term1 = self::factory()->term->create(
+			array(
+				'taxonomy' => 'wptests_tax',
+				'name'     => 'bar',
+			)
+		);
```
Comment (felixarntz) [inline tests/phpunit/tests/link/getEditTermLink.php]: Creating a new term here makes the test slower and is unnecessary. Instead, you could reuse one of the terms created for this entire test class (see `wpSetUpBeforeClass()` method). There's also the `get_term()` helper method above that makes this a one-liner.
```suggestion
		$term1 = $this->get_term( 'wptests_tax', true );
```

File: tests/phpunit/tests/link/getEditTermLink.php [hunk 2]
```diff
@@ -196,6 +196,25 @@ function ( $location, $term ) {
...
+
+		$actual   = get_edit_term_link( $term1 );
+		$expected = 'http://' . WP_TESTS_DOMAIN . '/wp-admin/term.php?taxonomy=wptests_tax&tag_ID=' . $term1 . '&post_type=post';
```
Comment (felixarntz) [inline tests/phpunit/tests/link/getEditTermLink.php]: Hard-coding the root URL is a bit error-prone here. Better to use the `admin_url()` function.
```suggestion
		$expected = admin_url( 'term.php?taxonomy=wptests_tax&tag_ID=' . $term1 . '&post_type=post' );
```

File: tests/phpunit/tests/link/getEditTermLink.php [hunk 3]
```diff
@@ -196,6 +196,25 @@ function ( $location, $term ) {
 
 		get_edit_term_link( $term, $taxonomy );
 	}
+	/**
+	 * This function checks if the return URL of `get_edit_term_link()` when called without taxonomy matches with the expected output or not.
+	 *
+	 * @ticket 61726
+	 *
+	 * @return void
```
Comment (felixarntz) [inline tests/phpunit/tests/link/getEditTermLink.php]: `@return void` is not commonly used in WordPress core.
```suggestion
```

**Review by felixarntz — COMMENTED** [review]
@Debarghya-Banerjee Thank you for the pull request!

It looks great overall. Just a few minor suggestions to polish the test.

File: tests/phpunit/tests/link/getEditTermLink.php (hunk 1 above)
Comment (Debarghya-Banerjee) [reply tests/phpunit/tests/link/getEditTermLink.php]: Fixed

File: tests/phpunit/tests/link/getEditTermLink.php (hunk 3 above)
Comment (Debarghya-Banerjee) [reply tests/phpunit/tests/link/getEditTermLink.php]: Fixed

File: tests/phpunit/tests/link/getEditTermLink.php (hunk 2 above)
Comment (Debarghya-Banerjee) [reply tests/phpunit/tests/link/getEditTermLink.php]: Fixed

**Comment by Debarghya-Banerjee** [comment]
Hi @felixarntz , Thanks you for the feedback, I have addressed them and made the necessary changes, assigning you again for re-review.

**Review by felixarntz — APPROVED** [review]
@Debarghya-Banerjee Thank you, this looks excellent!

**Comment by felixarntz** [comment]
Committed in https://core.trac.wordpress.org/changeset/58807

**Pull request closed without merging**